### PR TITLE
Refactor parquet PageReader to use PagedInputStream for decompression

### LINKED ISSUE #5714: Refactor PageReader::uncompressData to use PagedInputStream
### Description

Currently Parquet uses separate decompression functions in PageReader::uncompressData(). However DWRF and ORC manages the decompression by creating a PagedInputStream via ReaderBase::createDecompressedStream(). It would be better for Parquet to use PagedInputStream too. In `bool PagedInputStream::Next(const void** data, int32_t* size)`, the decompressor_ can handle all compression formats.

### PR DESCRIPTION
**Pull request by nmahadevuni**
Resolves #5714

### CODE REVIEW

File: velox/common/compression/CMakeLists.txt [hunk 1]
```diff
@@ -13,4 +13,4 @@
 # limitations under the License.
 
 add_library(velox_common_compression Compression.cpp LzoDecompressor.cpp)
-target_link_libraries(velox_common_compression Folly::folly)
+target_link_libraries(velox_common_compression velox_common_base velox_dwio_dwrf_proto velox_dwio_dwrf_common gtest Folly::folly)
```
Comment (Yuhta) [inline velox/common/compression/CMakeLists.txt:16]: The idea of this PR is great, but can we do it cleanly and step by step?
1. Decouple compression and `PagedInputStream` with `DWRF` dependencies
2. Use it in parquet `PageReader`

Comment (nmahadevuni) [reply velox/common/compression/CMakeLists.txt]: Do you suggest in two commits or two different PRs?

Comment (Yuhta) [reply velox/common/compression/CMakeLists.txt]: 2 different PR would be more easier to review

**Comment by nmahadevuni** [comment]
New PR for decoupling compression classes into common package is https://github.com/facebookincubator/velox/pull/5946

**Comment by nmahadevuni** [comment]
@yingsu00 @Yuhta Updated this PR. Please review.

File: velox/dwio/common/compression/Compression.cpp [hunk 2]
```diff
@@ -146,7 +150,12 @@ ZlibDecompressor::ZlibDecompressor(
   zstream_.opaque = Z_NULL;
   zstream_.next_out = Z_NULL;
   zstream_.avail_out = folly::to<uInt>(blockSize);
-  auto result = inflateInit2(&zstream_, -15);
+  int WINDOW_BITS = windowBits;
```
Comment (Yuhta) [inline velox/dwio/common/compression/Compression.cpp:153]: nit: camel case for variable name

File: velox/dwio/common/compression/Compression.h [hunk 3]
```diff
@@ -66,14 +66,20 @@ class Decompressor {
...
     facebook::velox::common::CompressionKind kind,
     std::unique_ptr<dwio::common::SeekableInputStream> input,
     uint64_t bufferSize,
     memory::MemoryPool& pool,
     const std::string& streamDebugInfo,
-    const dwio::common::encryption::Decrypter* decryptr = nullptr);
+    const dwio::common::encryption::Decrypter* decryptr = nullptr,
+    bool isRawUncompress = false,
+    size_t compressedLength = 0,
+    int zlibWindowBits = -15);
```
Comment (Yuhta) [inline velox/dwio/common/compression/Compression.h:82]: Can we create a struct to encapsulate the format specific options instead of passing them directly in the generic interface?  Passing something like this should be ok for now:

```c++
struct FormatOptions {
  union {
    int zlibWindowBits = -15;
  } zlib;
};
```

Comment (nmahadevuni) [reply velox/dwio/common/compression/Compression.h]: @Yuhta Updated the PR. Do you have any other comments?

File: velox/dwio/parquet/reader/PageReader.cpp [hunk 4]
```diff
@@ -225,84 +226,42 @@ const char* FOLLY_NONNULL PageReader::uncompressData(
...
-    }
-    case thrift::CompressionCodec::LZ4:
-    case thrift::CompressionCodec::LZO: {
-      return decompressLz4AndLzo(
-          pageData,
-          uncompressedData_,
-          compressedSize,
+  if (codec_ == thrift::CompressionCodec::LZ4 ||
+      codec_ == thrift::CompressionCodec::LZO) {
+    return decompressLz4AndLzo(
```
Comment (yingsu00) [inline velox/dwio/parquet/reader/PageReader.cpp:231]: Can we move this to the common decompressor? It already supports Lz4 and LZo. Is there any reason it's still here?

Comment (nmahadevuni) [reply velox/dwio/parquet/reader/PageReader.cpp:231]: Parquet seems to be using a different format for LZ4 and LZO and needs decoding headers, and uses multiple compression blocks. For ORC, it seems to be a single compression block. This needs some more understand to rework  before we can use PagedInputStream. Can work on it in a new PR.

Comment (yingsu00) [reply velox/dwio/parquet/reader/PageReader.cpp:231]: Ok, I created Issue #6105 for this.

File: velox/dwio/common/compression/Compression.h [hunk 5]
```diff
@@ -60,20 +60,40 @@ class Decompressor {
...
+
 /**
  * Create a decompressor for the given compression kind.
  * @param kind the compression type to implement
  * @param input the input stream that is the underlying source
  * @param bufferSize the maximum size of the buffer
  * @param pool the memory pool
+ * @param isRawUncompress specify whether to perform a raw uncompress
+ * @param compressedLength the compressed block length for raw uncompress
+ * @param zlibWindowBits window bits to use for zlib decompression
```
Comment (yingsu00) [inline velox/dwio/common/compression/Compression.h:85]: The comments were not updated yet

File: velox/dwio/common/compression/Compression.h [hunk 6]
```diff
@@ -60,20 +60,40 @@ class Decompressor {
...
     facebook::velox::common::CompressionKind kind,
     std::unique_ptr<dwio::common::SeekableInputStream> input,
     uint64_t bufferSize,
     memory::MemoryPool& pool,
     const std::string& streamDebugInfo,
-    const dwio::common::encryption::Decrypter* decryptr = nullptr);
+    const dwio::common::encryption::Decrypter* decryptr = nullptr,
+    bool isRawUncompress = false,
+    size_t compressedLength = 0,
+    CompressionOptions options = dwrfOrcCompressionOptions);
```
Comment (yingsu00) [inline velox/dwio/common/compression/Compression.h:96]: This comment is for Line 111 and 112. Would it make sense to wrap these levels in the options too?

File: velox/dwio/parquet/reader/PageReader.cpp [hunk 7]
```diff
@@ -225,84 +226,42 @@ const char* FOLLY_NONNULL PageReader::uncompressData(
...
+        codec_);
+  }
+
+  std::unique_ptr<dwio::common::SeekableInputStream> inputStream =
+      std::make_unique<dwio::common::SeekableArrayInputStream>(
+          pageData, compressedSize, 0);
+  auto streamDebugInfo =
+      fmt::format("Page Reader: Stream {}", inputStream_->getName());
+  struct dwio::common::compression::CompressionOptions parquetOptions;
+  parquetOptions.format.zlib.windowBits = 15;
```
Comment (yingsu00) [inline velox/dwio/parquet/reader/PageReader.cpp:246]: What do the 15 and -15 mean? I think it's better to create constants in common Decompressor class and reference them here. Creating meaningful name for these constants and add some comments would help reader understand their purpose.

File: velox/dwio/common/compression/PagedInputStream.h [hunk 8]
```diff
@@ -28,7 +28,9 @@ class PagedInputStream : public dwio::common::SeekableInputStream {
       memory::MemoryPool& memPool,
       std::unique_ptr<Decompressor> decompressor,
       const dwio::common::encryption::Decrypter* decrypter,
-      const std::string& streamDebugInfo)
+      const std::string& streamDebugInfo,
+      bool isRawUncompress = false,
```
Comment (yingsu00) [inline velox/dwio/common/compression/PagedInputStream.h:32]: isRawUncompress -> useRawDecompression

File: velox/dwio/parquet/reader/PageReader.cpp [hunk 9]
```diff
@@ -225,84 +226,42 @@ const char* FOLLY_NONNULL PageReader::uncompressData(
...
-    case thrift::CompressionCodec::LZO: {
-      return decompressLz4AndLzo(
-          pageData,
-          uncompressedData_,
-          compressedSize,
+  if (codec_ == thrift::CompressionCodec::LZ4 ||
+      codec_ == thrift::CompressionCodec::LZO) {
+    return decompressLz4AndLzo(
+        pageData,
+        uncompressedData_,
```
Comment (yingsu00) [inline velox/dwio/parquet/reader/PageReader.cpp:233]: Change all "uncompress" or "uncompressed" to "decompress" and "decompressed", including the function names. 
https://english.stackexchange.com/questions/56480/difference-between-uncompress-and-decompress
Uncompressed = not compressed; may never have been compressed in the first place. Just plain data is uncompressed data in every sense. To specifically refer to that which has undergone the process of decompressing, we would say decompressed data. Looking at it in another way, data before being compressed is uncompressed — that extracted from compressed format is decompressed data.

File: velox/dwio/common/compression/Compression.h [hunk 10]
```diff
@@ -60,20 +60,40 @@ class Decompressor {
...
+    Format() {
+      zlib.windowBits = -15;
+    }
+    struct {
+      int windowBits;
+    } zlib;
+  } format;
+};
+
+static struct CompressionOptions dwrfOrcCompressionOptions;
```
Comment (Yuhta) [inline velox/dwio/common/compression/Compression.h:75]: No `static` variable in header, just use the factory function above

File: velox/dwio/parquet/reader/PageReader.cpp (hunk 9 above)
Comment (nmahadevuni) [reply velox/dwio/parquet/reader/PageReader.cpp:233]: These variables were there before my changes. There are many instances of these. I would like to do this in another small PR, if it is ok.

**Comment by nmahadevuni** [comment]
@yingsu00 @Yuhta Addressed the comments. Please review again.

File: velox/dwio/common/compression/Compression.h [hunk 11]
```diff
@@ -60,20 +60,40 @@ class Decompressor {
   const std::string streamDebugInfo_;
 };
 
+struct CompressionOptions {
+  CompressionOptions() {}
```
[the remark this answers is not in the document]
Comment (Yuhta) [reply velox/dwio/common/compression/Compression.h:71]: Can we remove the constructors (`CompressionOptions()`, `Format()`) all together?

File: velox/dwio/parquet/reader/PageReader.cpp (hunk 9 above)
Comment (yingsu00) [reply velox/dwio/parquet/reader/PageReader.cpp:233]: @nmahadevuni Will you be able to add another commit to rename these in this PR?

Comment (nmahadevuni) [reply velox/dwio/parquet/reader/PageReader.cpp]: I renamed all uncompress to decompress except the auto generated thrift types. Added as a new commit.

**Comment by nmahadevuni** [comment]
@yingsu00 @Yuhta Addressed the latest comments. Please review!

File: velox/dwio/common/compression/PagedInputStream.h [hunk 12]
```diff
@@ -28,7 +28,9 @@ class PagedInputStream : public dwio::common::SeekableInputStream {
       memory::MemoryPool& memPool,
       std::unique_ptr<Decompressor> decompressor,
       const dwio::common::encryption::Decrypter* decrypter,
-      const std::string& streamDebugInfo)
+      const std::string& streamDebugInfo,
+      bool useRawDecompression = false,
```
Comment (xiaoxmeng) [inline velox/dwio/common/compression/PagedInputStream.h:32]: Can we move streamDebugInfo to the last input?

File: velox/dwio/common/compression/Compression.h [hunk 13]
```diff
@@ -60,20 +67,70 @@ class Decompressor {
...
+}
+
+static CompressionOptions getDwrfOrcDecompressionOptions() {
+  CompressionOptions options;
+  options.format.zlib.windowBits = Compressor::DWRF_ORC_ZLIB_WINDOW_BITS;
+  return options;
+}
+
+static CompressionOptions getParquetDecompressionOptions() {
+  CompressionOptions options;
```
Comment (xiaoxmeng) [inline velox/dwio/common/compression/Compression.h:109]:
```
const CompressionOptions& options;
```

File: velox/dwio/common/compression/Compression.h [hunk 14]
```diff
@@ -60,20 +67,70 @@ class Decompressor {
...
     facebook::velox::common::CompressionKind kind,
     std::unique_ptr<dwio::common::SeekableInputStream> input,
     uint64_t bufferSize,
     memory::MemoryPool& pool,
     const std::string& streamDebugInfo,
-    const dwio::common::encryption::Decrypter* decryptr = nullptr);
+    const dwio::common::encryption::Decrypter* decryptr = nullptr,
+    bool useRawDecompression = false,
+    size_t compressedLength = 0,
+    CompressionOptions options = getDwrfOrcDecompressionOptions());
```
Comment (xiaoxmeng) [inline velox/dwio/common/compression/Compression.h:133]:
```
const CompressionOptions& options;
```

File: velox/dwio/common/compression/Compression.h [hunk 15]
```diff
@@ -60,20 +67,70 @@ class Decompressor {
...
 std::unique_ptr<dwio::common::SeekableInputStream> createDecompressor(
     facebook::velox::common::CompressionKind kind,
     std::unique_ptr<dwio::common::SeekableInputStream> input,
     uint64_t bufferSize,
     memory::MemoryPool& pool,
     const std::string& streamDebugInfo,
-    const dwio::common::encryption::Decrypter* decryptr = nullptr);
+    const dwio::common::encryption::Decrypter* decryptr = nullptr,
+    bool useRawDecompression = false,
+    size_t compressedLength = 0,
```
Comment (xiaoxmeng) [inline velox/dwio/common/compression/Compression.h:132]:
```
const CompressionOptions& options;
```

**Review by xiaoxmeng — COMMENTED** [review]
@nmahadevuni thanks for the change. Can we add tests to verify the passed options have been handled properly. Thanks!

File: velox/dwio/common/compression/Compression.cpp [hunk 16]
```diff
@@ -146,7 +150,12 @@ ZlibDecompressor::ZlibDecompressor(
   zstream_.opaque = Z_NULL;
   zstream_.next_out = Z_NULL;
   zstream_.avail_out = folly::to<uInt>(blockSize);
-  auto result = inflateInit2(&zstream_, -15);
+  int zlibWindowBits = windowBits;
+  constexpr int GZIP_DETECT_CODE = 32;
```
[the remark this answers is not in the document]
Comment (Yuhta) [reply velox/dwio/common/compression/Compression.cpp:154]: There is no difference between the two, this variable is inlined and does not exist in runtime anyway

File: velox/dwio/common/compression/Compression.cpp [hunk 17]
```diff
@@ -499,10 +515,34 @@ std::unique_ptr<dwio::common::SeekableInputStream> createDecompressor(
...
+        // When file is not encrypted, we can use zlib streaming codec to avoid
+        // copying data
+        return std::make_unique<ZlibDecompressionStream>(
+            std::move(input),
+            blockSize,
+            pool,
+            streamDebugInfo,
+            options.format.zlib.windowBits,
+            true,
+            useRawDecompression,
```
[the remark this answers is not in the document]
Comment (nmahadevuni) [reply velox/dwio/common/compression/Compression.cpp:541]: No, this is for PagedInputStream, not specific to any compression format.

File: velox/dwio/common/compression/Compression.cpp [hunk 18]
```diff
@@ -137,7 +139,9 @@ class ZlibDecompressor : public Decompressor {
 
 ZlibDecompressor::ZlibDecompressor(
     uint64_t blockSize,
-    const std::string& streamDebugInfo)
+    const std::string& streamDebugInfo,
+    int windowBits,
```
[the remark this answers is not in the document]
Comment (nmahadevuni) [reply velox/dwio/common/compression/Compression.cpp:143]: Moved windowBits. isGzip has a default value. Cannot move it.

File: velox/dwio/common/compression/Compression.cpp [hunk 19]
```diff
@@ -337,9 +346,13 @@ class ZlibDecompressionStream : public PagedInputStream,
       std::unique_ptr<dwio::common::SeekableInputStream> inStream,
       uint64_t blockSize,
       MemoryPool& pool,
-      const std::string& streamDebugInfo)
-      : PagedInputStream{std::move(inStream), pool, streamDebugInfo},
-        ZlibDecompressor{blockSize, streamDebugInfo} {}
+      const std::string& streamDebugInfo,
+      int windowBits,
```
[the remark this answers is not in the document]
Comment (nmahadevuni) [reply velox/dwio/common/compression/Compression.cpp:350]: it has default arguments, streamDebugInfo cannot be moved to last.

File: velox/dwio/common/compression/Compression.cpp [hunk 20]
```diff
@@ -436,9 +449,7 @@ std::unique_ptr<BufferedOutputStream> createCompressor(
     CompressionKind kind,
     CompressionBufferPool& bufferPool,
     DataBufferHolder& bufferHolder,
-    uint32_t compressionThreshold,
-    int32_t zlibCompressionLevel,
-    int32_t zstdCompressionLevel,
+    CompressionOptions options,
```
[the remark this answers is not in the document]
Comment (nmahadevuni) [reply velox/dwio/common/compression/Compression.cpp:452]: the last argument is a default argument, this cannot be moved to last.

File: velox/dwio/common/compression/PagedInputStream.h (hunk 12 above)
Comment (nmahadevuni) [reply velox/dwio/common/compression/PagedInputStream.h:32]: It has default arguments, streamDebugInfo can't be moved to last.

File: velox/dwio/common/compression/Compression.h (hunk 13 above)
Comment (nmahadevuni) [reply velox/dwio/common/compression/Compression.h:109]: can't declare it const reference. needs initialization.

**Comment by nmahadevuni** [comment]
@xiaoxmeng Addressed the comments. Please review again. @Yuhta @yingsu00

File: velox/dwio/parquet/reader/PageReader.cpp [hunk 21]
```diff
@@ -125,37 +125,37 @@ const char* PageReader::readBytes(int32_t size, BufferPtr& copy) {
 
 const char* FOLLY_NONNULL decompressLz4AndLzo(
     const char* compressedData,
-    BufferPtr& uncompressedData,
+    BufferPtr& decompressedData,
     uint32_t compressedSize,
-    uint32_t uncompressedSize,
+    uint32_t decompressedSize,
```
Comment (yingsu00) [inline velox/dwio/parquet/reader/PageReader.cpp:130]: This should be uncompressedSize, which means the size before the data was compressed

File: velox/dwio/parquet/reader/PageReader.cpp [hunk 22]
```diff
@@ -125,37 +125,37 @@ const char* PageReader::readBytes(int32_t size, BufferPtr& copy) {
...
     inputLength -= dwio::common::INT_BYTE_SIZE;
-    uint32_t remainingOutputSize = uncompressedSize - decompressedTotalLength;
-    if (remainingOutputSize < uncompressedBlockLength) {
+    uint32_t remainingOutputSize = decompressedSize - decompressedTotalLength;
+    if (remainingOutputSize < decompressedBlockLength) {
       VELOX_FAIL(
-          "{} uncompress failed, remainingOutputSize is less then "
-          "uncompressedBlockLength, remainingOutputSize: {}, "
-          "uncompressedBlockLength: {}",
+          "{} decompress failed, remainingOutputSize is less then "
```
Comment (yingsu00) [inline velox/dwio/parquet/reader/PageReader.cpp:154]: decompression failed

File: velox/dwio/parquet/reader/PageReader.cpp [hunk 23]
```diff
@@ -165,7 +165,7 @@ const char* FOLLY_NONNULL decompressLz4AndLzo(
       // Check that input length should not be negative.
       if (inputLength < sizeof(uint32_t)) {
         VELOX_FAIL(
-            "{} uncompress failed, input len is to small: {}",
+            "{} decompress failed, input len is to small: {}",
```
Comment (yingsu00) [inline velox/dwio/parquet/reader/PageReader.cpp:168]: decompression failed

File: velox/dwio/common/compression/Compression.h [hunk 24]
```diff
@@ -60,20 +67,46 @@ class Decompressor {
...
+  uint32_t compressionThreshold;
+};
+
 /**
  * Create a decompressor for the given compression kind.
  * @param kind the compression type to implement
  * @param input the input stream that is the underlying source
  * @param bufferSize the maximum size of the buffer
  * @param pool the memory pool
+ * @param useRawDecompression specify whether to perform raw decompression
```
Comment (yingsu00) [inline velox/dwio/common/compression/Compression.h:96]: Can you please make the comments order the same as the real arguments?

File: velox/dwio/common/compression/Compression.h [hunk 25]
```diff
@@ -87,10 +120,8 @@ std::unique_ptr<BufferedOutputStream> createCompressor(
     facebook::velox::common::CompressionKind kind,
```
Comment (yingsu00) [inline velox/dwio/common/compression/Compression.h:120]: This comment is for line 117: will you please update the comments accordingly?

File: velox/dwio/dwrf/common/Compression.h [hunk 26]
```diff
@@ -26,8 +26,27 @@
 
 namespace facebook::velox::dwrf {
 
+using namespace dwio::common::compression;
+
 constexpr uint8_t PAGE_HEADER_SIZE = 3;
 
+FOLLY_ALWAYS_INLINE const CompressionOptions getDwrfOrcCompressionOptions(
```
Comment (yingsu00) [inline velox/dwio/dwrf/common/Compression.h:33]: Why use FOLLY_ALWAYS_INLINE? This function is just called once when creating the compressor.

File: velox/dwio/dwrf/common/Compression.h [hunk 27]
```diff
@@ -38,19 +57,54 @@ constexpr uint8_t PAGE_HEADER_SIZE = 3;
...
+  return createCompressor(
+      kind,
+      bufferPool,
+      bufferHolder,
       PAGE_HEADER_SIZE,
+      dwrfOrcCompressionOptions,
       encrypter);
 }
 
+FOLLY_ALWAYS_INLINE const CompressionOptions getDwrfOrcDecompressionOptions() {
```
Comment (yingsu00) [inline velox/dwio/dwrf/common/Compression.h:79]: Why FOLLY_ALWAYS_INLINE?

File: velox/dwio/parquet/reader/CMakeLists.txt [hunk 28]
```diff
@@ -36,4 +36,5 @@ target_link_libraries(
   arrow
   Snappy::snappy
   thrift
-  zstd::zstd)
+  zstd::zstd
+  ${Protobuf_LIBRARIES})
```
Comment (yingsu00) [inline velox/dwio/parquet/reader/CMakeLists.txt:40]: Is this line related and necessary?

File: velox/dwio/parquet/reader/PageReader.h [hunk 29]
```diff
@@ -28,6 +30,8 @@
 
 namespace facebook::velox::parquet {
 
+using namespace dwio::common::compression;
```
Comment (yingsu00) [inline velox/dwio/parquet/reader/PageReader.h:33]: By convention, the header files do not use "using namespace". Can you please just inline these prefixes?

File: velox/dwio/dwrf/common/Compression.h (hunk 26 above)
Comment (nmahadevuni) [reply velox/dwio/dwrf/common/Compression.h:33]: removed

File: velox/dwio/dwrf/common/Compression.h (hunk 27 above)
Comment (nmahadevuni) [reply velox/dwio/dwrf/common/Compression.h:79]: removed

File: velox/dwio/parquet/reader/CMakeLists.txt (hunk 28 above)
Comment (nmahadevuni) [reply velox/dwio/parquet/reader/CMakeLists.txt:40]: This is necessary since we are now having references google::protobuf::io::ZeroCopyOutputStream via BufferedOutputStream from dwio/common/compression/Compression.h -> velox/dwio/common/OutputStream.h

File: velox/dwio/common/compression/Compression.h [hunk 30]
```diff
@@ -60,37 +67,62 @@ class Decompressor {
...
+};
+
 /**
  * Create a decompressor for the given compression kind.
  * @param kind the compression type to implement
  * @param input the input stream that is the underlying source
  * @param bufferSize the maximum size of the buffer
  * @param pool the memory pool
+ * @param options compression options to use
+ * @param useRawDecompression specify whether to perform raw decompression
```
Comment (yingsu00) [inline velox/dwio/common/compression/Compression.h:97]: This parameter is only for GZIp and ZLIB, can you mention it in the comment?

File: velox/dwio/common/compression/Compression.h [hunk 31]
```diff
@@ -60,37 +67,62 @@ class Decompressor {
...
+      int32_t compressionLevel;
+    } zstd;
+  } format;
+
+  uint32_t compressionThreshold;
+};
+
 /**
  * Create a decompressor for the given compression kind.
  * @param kind the compression type to implement
```
Comment (yingsu00) [inline velox/dwio/common/compression/Compression.h:92]: Can you please use Upper case for the first characters? E.g. 
@param kind The compression type to implement

ditto

File: velox/dwio/dwrf/common/Compression.h [hunk 32]
```diff
@@ -38,19 +57,54 @@ constexpr uint8_t PAGE_HEADER_SIZE = 3;
...
 
+static const CompressionOptions getDwrfOrcDecompressionOptions() {
+  CompressionOptions options;
+  options.format.zlib.windowBits = Compressor::DWRF_ORC_ZLIB_WINDOW_BITS;
+  return options;
+}
+
+/**
+ * Create a decompressor for the given compression kind.
+ * @param kind the compression type to implement
```
Comment (yingsu00) [inline velox/dwio/dwrf/common/Compression.h:87]: Upper case for the first chars

**Review by yingsu00 — COMMENTED** [review]
@nmahadevuni Mostly look good! Just a couple of nits...

File: velox/dwio/common/compression/Compression.h (hunk 30 above)
Comment (nmahadevuni) [reply velox/dwio/common/compression/Compression.h:97]: It is currently for Gzip and Zlib, but can be used for any format. Do we want to change the comment?

Comment (nmahadevuni) [reply velox/dwio/common/compression/Compression.h:97]: Also compressionThreshold is not specific to these formats.

**Comment by nmahadevuni** [comment]
@ying, can you rerun the failed test? It is not related.

**Review by yingsu00 — APPROVED** [review]

File: velox/dwio/parquet/reader/CMakeLists.txt (hunk 28 above)
Comment (baibaichen) [reply velox/dwio/parquet/reader/CMakeLists.txt:40]: ## undefined reference to `typeinfo for google::protobuf::io::ZeroCopyOutputStream' after this PR

I got the following link issue
```
/usr/bin/ld: velox/dwio/common/compression/libvelox_dwio_common_compression.a(Compression.cpp.o): in function `google::protobuf::io::ZeroCopyOutputStream::ZeroCopyOutputStream()':
/home/chang/SourceCode/velox/cmake-build-debug-system/_deps/protobuf-src/src/google/protobuf/io/zero_copy_stream.h:193: undefined reference to `vtable for google::protobuf::io::ZeroCopyOutputStream'
/usr/bin/ld: velox/dwio/common/compression/libvelox_dwio_common_compression.a(Compression.cpp.o): in function `google::protobuf::io::ZeroCopyOutputStream::~ZeroCopyOutputStream()':
/velox/cmake-build-debug-system/_deps/protobuf-src/src/google/protobuf/io/zero_copy_stream.h:194: undefined reference to `vtable for google::protobuf::io::ZeroCopyOutputStream'
/usr/bin/ld: velox/dwio/common/libvelox_dwio_common.a(OutputStream.cpp.o):(.[data.rel.ro](http://data.rel.ro/)._ZTIN8facebook5velox4dwio6common20BufferedOutputStreamE[_ZTIN8facebook5velox4dwio6common20BufferedOutputStreamE]+0x10): undefined reference to `typeinfo for google::protobuf::io::ZeroCopyOutputStream'
```
Here are my  cmake option

```
-DVELOX_DEPENDENCY_SOURCE=BUNDLED
-DVELOX_ENABLE_EXAMPLES=ON
-DVELOX_ENABLE_PARQUET=ON
-DVELOX_BUILD_TESTING=ON
-DVELOX_ENABLE_BENCHMARKS=ON
-DVELOX_ENABLE_BENCHMARKS_BASIC=ON
```

I checked probobuf cmake  file and make sure **disbale RTTI is OFF**

Not sure how to fix it.

Comment (majetideepak) [reply velox/dwio/parquet/reader/CMakeLists.txt:40]: We should have fixed the protobuf header leak instead of propagating this dependency. I will open a PR to fix this.
